File: 3616-make-array-elements-equal-to-zero/3616-make-array-elements-equal-to-zero.java
class Solution {
    public int countValidSelections (int [] nums) {
        int count = 0, left = 0;
        int right = Arrays.stream (nums).sum ();
        for (int i = 0; i < nums.length; i++) {
            left += nums [i];
            right -= nums [i];
            if (nums[i] != 0) continue;
            if (left == right) count += 2;
            if (Math.abs (left - right) == 1) count++;
        }
        return count;
    }
}

File: 3616-make-array-elements-equal-to-zero/3616-make-array-elements-equal-to-zero.c
int countValidSelections (int* nums, int numsSize) {
    int count = 0, left = 0, right = 0;
    for (int i = 0; i < numsSize; i++) right += nums [i];
    for (int i = 0; i < numsSize; i++) {
        left += nums [i];
        right -= nums [i];
        if (nums [i] != 0) continue;
        if (left == right) count += 2;
        if (abs (left - right) == 1) count++;
    }
    return count;
}